src/studio/nuxeo-online-services-portal/External Source Repository.md

```diff
@@ -0,0 +1,57 @@
+description: Nuxeo Studio Integration with External Git Repositories
+review:
+    comment: ''
+    date: ''
+    status: ok
+toc: true
+tree_item_index: 400
+---
+
+
+
+
+
+# Nuxeo Studio Integration with External Git Repositories
+
+Nuxeo Studio, a powerful customization tool for the Nuxeo Platform, has recently introduced a groundbreaking feature enabling seamless integration with external Git repositories. This enhancement allows organizations to leverage the collaborative power of Git while managing their Nuxeo Studio configurations, thus streamlining the development and deployment processes.
+
+## Integration Guide: Linking Nuxeo Studio Project to External Git Repository
+
+*NOTE:-For External Git access, you need to contact support.*
+
+### Step 1: Navigate to Settings to enable Git external repository
+
+1. Within your Studio project, navigate to the top-right corner and locate the "Settings" button.
+
+ ![]({{file name='settings.png'}} ?w=650,border=true)
+
+
+2. From the connect dashboard, locate your desired Studio project and click on it to enter.
+3. Once selected, click on the "Edit" button to enter the editing mode for your chosen studio project.
+
+### Step 2: Configuring Git Credentials
+
+1. After entering edit mode, navigate to the "source repo" tab within your sandbox configuration.
+
+   ![]({{file name='sourcerepo.png'}} ?w=650,border=true)
+ 
+3. Here, you'll find options to configure Git credentials for authentication with the external repository.
+4. Enable the toggle button within the source Repo section and entering your Git credentials and then save it.
+
+Once the enablement process is complete, your Studio project is now linked to the specified external Git repository associated with your studio project.
+
+ ![]({{file name='linkedrepo.png'}} ?w=650,border=true)
+
+---
+### Use of Wip branch for external git 
+The WIP branch is typically used when you're actively developing a feature or making changes to the codebase that aren't ready to be merged into the main codebase yet. It provides a sandbox environment where you can make changes, test them, and collaborate with other team members without impacting the stability of the main branch (often called "master" or "main").
+
+Using a WIP branch allows you to keep the main branch clean and stable while still giving you the flexibility to iterate on your changes without disrupting the workflow of others on your team. Once the feature or changes are complete and tested, they can be merged back into the main branch through a process called "merge" or "pull request," depending on the version control system you're using.
+
+NOTE:-Your created WIP branch name should be wipm/{{your USERNAME}}
+
+
+### Overview
+By automating the process of merging changes from Nuxeo Studio's simple mode directly into the master branch of the linked Git repository, users can streamline their workflow and ensure efficient version control management while maintaining flexibility and ease of use.
```

Comment:
```suggestion
## The WIP branch concept
While editing your project using the Studio UI, you will notice the creation of branches with the following pattern:

wip/{{USERNAME}}_{{USERHASH}}/{{FEATURE_BRANCH}}

These branches are used as workspaces for each user working on a specific feature branch and are not meant to be edited "manually" unless you know what you are doing, otherwise the corresponding user may not be able to load the UI.
Nevertheless, playing with these branches can sometimes be useful and allows you to control what is effectively loaded when doing hotreloads for example !

## Commit preferences
The recommended Commit preferences when External Source Repository is configured is "Advanced" or "Intermediate", the "Simple" mode will have degraded performances when hitting the "Save" button has the automatic pull/merge will be triggered every time.
```